src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -183,17 +191,46 @@ public static bool UpdateInstallerNodes(InstallerManifest installerManifest, IEn
             // Update previous installers with parsed data from downloaded packages
             foreach (var newInstaller in newInstallers)
             {
-                // Find a match for the current installer in the list of installers remaining to be updated.
-                var matchingExistingInstaller = existingInstallers.SingleOrDefault(i =>
-                    (i.InstallerType ?? installerManifest.InstallerType) == newInstaller.InstallerType &&
+                var archGuess = GetArchFromUrl(newInstaller.InstallerUrl);
+
```

Comment:
Log a warning here if our archGuess != newInstaller.Architecture, this'll let the user know we're overriding what was detected in the binary, and possibly spur them to fix the binary arch too...

Reply:
Added a log message that tells the user the detected arch in the binary differs from what what detected in the URL. Let me know what you think and if it needs more clarification.